### ISSUE TITLE
Jira tickets get created for duplicate findings

### ISSUE THREAD

**Issue by michaelgibson** [DefectDojo/django-DefectDojo#3944]
It looks like the Deduplication and Jira celery tasks are committed to the queue asynchronously.
The deduplication task marks a dupe finding duplicate, but because the jira task has finding data that preceeds that it still sees  'duplicate' field is always False.
Additionally, when it goes to save the finding, it overwrites the duplicate field as false even if the dedup task marked it as True.
https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/jira_link/helper.py#L550

I've worked around this by telling the jira helper code to do a double check on the live status of the finding before proceeding.
https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/jira_link/helper.py#L469

`  
finding = Finding.objects.get(id=find.id)

  if finding.duplicate:
    logger.debug('Finding has been marked duplicate. Skipping Jira ticket creation.')
    return
`

**Comment by valentijnscholten** [comment]
This indicates a timing issue, which may well be because we always let these tasks run in parallel. On fast system I think it all goes well, but there's a risk of the above indeed. The solution would be to make the tasks sequential / ordered. The solution proposed above might help, but still has the same risk of creating a JIRA finding before the finding is deduplicated.

Are you use `push_all_issues` or `push_to_jira` or just using the UI to save a finding?

**Comment by valentijnscholten** [comment]
The `.save()` calls are not needed anymore in the jira helper, so I removed those in 3953. Won't fix the timing issue fully, but reduces the impact quite a bit.

**Comment by michaelgibson** [comment]
> This indicates a timing issue, which may well be because we always let these tasks run in parallel. On fast system I think it all goes well, but there's a risk of the above indeed. The solution would be to make the tasks sequential / ordered. The solution proposed above might help, but still has the same risk of creating a JIRA finding before the finding is deduplicated.
> 
> Are you use `push_all_issues` or `push_to_jira` or just using the UI to save a finding?

We mainly specify `push_to_jira` though the API at the Finding or import-scan level depending on tool.
